File: source/src/main/webapp/js/pages/transversalobject/ApplicationObject.js
/*
 * Cerberus  Copyright (C) 2013  vertigo17
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This file is part of Cerberus.
 *
 * Cerberus is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Cerberus is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Cerberus.  If not, see <http://www.gnu.org/licenses/>.
 */

function initPageModal(page) {
    displayModalLabel();

    // handle the click for specific action buttons
    $("#addApplicationObjectButton").click(function(){
        addApplicationObjectModalSaveHandler(page)
    });

    //clear the modals fields when closed
    $('#addApplicationObjectModal').on('hidden.bs.modal', addApplicationObjectModalCloseHandler);
}

function displayModalLabel() {
    var doc = new Doc();

    $("[name='createApplicationObjectField']").html(doc.getDocLabel("page_applicationObject", "createapplicationobjectfield"));
    $("[name='applicationField']").html(doc.getDocLabel("page_applicationObject", "applicationfield"));
    $("[name='objectField']").html(doc.getDocLabel("page_applicationObject", "objectfield"));
    $("[name='valueField']").html(doc.getDocLabel("page_applicationObject", "valuefield"));
    $("[name='screenshotfilenameField']").html(doc.getDocLabel("page_applicationObject", "screenshotfilenamefield"));
    $("[name='buttonClose']").html(doc.getDocLabel("page_applicationObject", "button_close"));
    $("[name='buttonAdd']").html(doc.getDocLabel("page_applicationObject", "button_add"));

}

function addApplicationObjectModalSaveHandler(page) {
    clearResponseMessage($('#addApplicationObjectModal'));
    var formAdd = $("#addApplicationObjectModal #addApplicationObjectModalForm :input");
    var file = $("#addApplicationObjectModal input[type=file]");
    // Get the header data from the form
    var sa = formAdd.serializeArray();
    var formData = new FormData();
    var data = {}
    for (var i in sa) {
        formData.append(sa[i].name, sa[i].value);
    }

    formData.append("file",file.prop("files")[0]);
    showLoaderInModal('#addApplicationObjectModal');
    var jqxhr = $.ajax({
        type: "POST",
        url: "CreateApplicationObject",
        data: formData,
        processData: false,
        contentType: false
    });
    $.when(jqxhr).then(function (data) {
        hideLoaderInModal('#addApplicationObjectModal');
//        console.log(data.messageType);
        if (getAlertType(data.messageType) === 'success') {
            if(page == "applicationObject") {
                var oTable = $("#applicationObjectsTable").dataTable();
                oTable.fnDraw(true);
            }else if(page == "testCaseScript"){
                $("div.step-action .content div.row.form-inline span:nth-child(n+2) input").trigger("change");
            }
            showMessage(data);
            $('#addApplicationObjectModal').modal('hide');

        } else {
            showMessage(data, $('#addApplicationObjectModal'));
        }
    }).fail(handleErrorAjaxAfterTimeout);
}

function addApplicationObjectModalCloseHandler() {
    // reset form values
    $('#addApplicationObjectModal #addApplicationObjectModalForm')[0].reset();
    // remove all errors on the form fields
    $(this).find('div.has-error').removeClass("has-error");
    // clear the response messages of the modal
    clearResponseMessage($('#addApplicationObjectModal'));
}

function addApplicationObjectModalClick(event, object, application) {
    clearResponseMessageMainPage();

    $('#addApplicationObjectModal #application').empty();
    displayApplicationList("application","",application);

    if(object != undefined){
        $("[name='object']").val(object);
    }

    // When creating a new applicationObject, System takes the default value of the 
    // system already selected in header.
    var formAdd = $('#addApplicationObjectModal');
    // Default to NONE to Application.
    formAdd.find("#type").val("NONE");

    $('#addApplicationObjectModal').modal('show');
}

File: source/src/main/java/org/cerberus/crud/entity/ApplicationObject.java
/**
 * Cerberus Copyright (C) 2013 - 2017 cerberustesting
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This file is part of Cerberus.
 *
 * Cerberus is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Cerberus is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Cerberus.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.cerberus.crud.entity;

/**
 * @author vertigo
 */
public class ApplicationObject {

    private int ID;
    private String application;
    private String object;
    private String value;
    private String screenshotfilename;
    private String usrcreated;
    private String datecreated;
    private String usrmodif;
    private String datemodif;



    public int getID() {
        return ID;
    }

    public void setID(int ID) {
        this.ID = ID;
    }

    public String getApplication() {
        return application;
    }

    public void setApplication(String application) {
        this.application = application;
    }

    public String getObject() {
        return object;
    }

    public void setObject(String object) {
        this.object = object;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public String getScreenShotFileName() {
        return screenshotfilename;
    }

    public void setScreenShotFileName(String screenshotfilename) {
        this.screenshotfilename = screenshotfilename;
    }

    public String getUsrCreated() { return usrcreated; }

    public void setUsrCreated(String usrcreated) {
        this.usrcreated = usrcreated;
    }

    public String getDateCreated() { return datecreated; }

    public void setDateCreated(String datecreated) {
        this.datecreated = datecreated;
    }

    public String getUsrModif() { return usrmodif; }

    public void setUsrModif(String usrmodif) {
        this.usrmodif = usrmodif;
    }

    public String getDateModif() { return datemodif; }

    public void setDateModif(String datemodif) {
        this.datemodif = datemodif;
    }

    @Override
    public String toString() {
        return application + " " + object;
    }
}
